stablehlo/tests/vhlo/stablehlo_legalize_to_vhlo.mlir

```diff
@@ -2729,6 +2729,22 @@ func.func @type_ui64(%arg0: tensor<ui64>, %arg1: tensor<ui64>) -> tensor<ui64> {
   func.return %0 : tensor<ui64>
 }
 
+// CHECK-LABEL: "type_f8E3M4"
+// CHECK-NEXT: (%[[ARG0:.*]]: {{.*}}, %[[ARG1:.*]]: {{.*}})
+func.func @type_f8E3M4(%arg0: tensor<f8E3M4>, %arg1: tensor<f8E3M4>) -> tensor<f8E3M4> {
```

Comment:
Compatibility impl and backward compat test lgtm, can we add a forward compat test that this will fail to downgrade to StableHLO versions that don't support it, similar to:
https://github.com/openxla/stablehlo/blob/main/stablehlo/tests/vhlo/vhlo_to_version_downgrade_invalid.0_10_0.mlir

Should be called `vhlo_to_version_downgrade_invalid.1_6_0.mlir` and fail when trying to downgrade from 1.7.0 to 1.6.0

Reply:
I added `vhlo_to_version_downgrade_invalid.1_6_0.mlir` which is a clone of `vhlo_to_version_downgrade_invalid.1_1_0.mlir` (tests for i2 and ui2)